src/nyc_trees/js/src/group_list.js

```diff
@@ -1,4 +1,41 @@
 "use strict";
 
+var $ = require('jquery');
+
+var attrs = {name: 'data-group-name',
+             description: 'data-group-description'},
+    dom = {searchInput: '[data-class="group-search"]',
+           viewAll: '[data-class="group-view-all"]',
+           groups: '[' + attrs.name + ']'},
+    $groups = $(dom.groups);
+
 // Needed for Follow button event handlers.
 require('./followGroupButton');
+
+$(dom.viewAll).on('click', function () {
+    $groups.show();
+    $(dom.viewAll).hide();
+});
+
+$(dom.searchInput).on('input', function (e) {
+    var query = $(e.target).val();
+
+    $(dom.viewAll).hide();
+
+    if (query === '') {
+        $groups.show();
+    } else {
+        $.each($groups, function(__, group) {
+            var $group = $(group),
+                name = $group.attr(attrs.name),
+                description = $group.attr(attrs.description);
+
+            if (name.indexOf(query) !== -1 ||
```

Comment:
This is already done in the template:
https://github.com/azavea/nyc-trees/pull/273/files#diff-55927da6a21d8a4a5666708880315de6R21